### PR TITLE
Tests for the final ui version

### PR DESCRIPTION
**Pull request by arushigarg18**

### CODE REVIEW

File: ui/tests_ui_vfinal/master_test.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  testTableSelectionFunctions();
+  testDetectDateFunctions();
+  testQueryProcessingFunctions();
+  testDataFormattingFunctions();
+  testCreateQueryFunctions();
+  testErrorCheckFunctions();
+}
+
+/**
+ * Compare the values received and check if they are equal
```
Comment (SowmithManepalli) [inline ui/tests_ui_vfinal/master_test.js:16]: Add helper function in the comment.

File: ui/tests_ui_vfinal/test_errorchecks.js [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+  return true;
+}
+
+/**
+ * Test: checkRangeValidity(a1Notation)
+ * When the A1Notation given is valid
+ */
+function testCheckRangeValidity1() {
+  // Call the function to test
+  var generatedOutput = checkRangeValidity('C5:20');
```
Comment (SowmithManepalli) [inline ui/tests_ui_vfinal/test_errorchecks.js:171]: you could actually test these with a lot of ranges, just to be 100% sure of the logic, just add a TODO() here.

File: ui/tests_ui_vfinal/test_errorchecks.js [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+  // Checking if generated output is same as expected output
+  assertEquals(expectedOutput, generatedOutput, 'checkHeaderValidity2');
+}  
+
+/**
+ * Check if the specified header row is valid
+ * @param {string} headerRowValue The header row to be checked for validity
+ * @return {boolean} True indictes valid and false indicates invalid 
+ */
+function checkHeaderValidity(headerRowValue) {
```
Comment (SowmithManepalli) [inline ui/tests_ui_vfinal/test_errorchecks.js:279]: check headerrow validity ?

File: ui/tests_ui_vfinal/test_errorchecks.js (hunk 2 above)
Comment (arushigarg18) [reply ui/tests_ui_vfinal/test_errorchecks.js:171]: Todo stating ?

Comment (SowmithManepalli) [reply ui/tests_ui_vfinal/test_errorchecks.js:171]: TODO: Test valid range with more cases.

File: ui/tests_ui_vfinal/test_errorchecks.js (hunk 3 above)
Comment (arushigarg18) [reply ui/tests_ui_vfinal/test_errorchecks.js:279]: Will update

**Pull request merged**